firefox-ios/Storage/DiskImageStore.swift

```diff
@@ -82,12 +82,32 @@ public actor DefaultDiskImageStore: DiskImageStore {
 
     public func saveImageForKey(_ key: String, image: UIImage) async throws {
         let imageURL = URL(fileURLWithPath: filesDir).appendingPathComponent(key)
-        if let data = image.jpegData(compressionQuality: quality) {
-            try data.write(to: imageURL, options: .noFileProtection)
-            keys.insert(key)
-        } else {
+        let scaledImage = scaleImageFrom3xTo1x(image)
+
+        guard let data = scaleImageFrom3xTo1x(image).jpegData(compressionQuality: quality) else {
```

Comment:
```suggestion
        guard let data = scaledImage.jpegData(compressionQuality: quality) else {
```

(unused var line 85)